### ISSUE TITLE
rr:constant and rr:parentTriplesMap without join conditions

### ISSUE THREAD

**Issue by eugenesiow** [oeg-upm/morph-rdb#21]
Hi, I am working with LinkedSensorData and morph and have encountered the following problems.

This is my database table schema

```
describe _4UT01;
+------------------+---------------+------+-----+-------------------+-----------------------------+
| Field            | Type          | Null | Key | Default           | Extra                       |
+------------------+---------------+------+-----+-------------------+-----------------------------+
| time             | timestamp     | NO   |     | CURRENT_TIMESTAMP | on update CURRENT_TIMESTAMP |
| Precipitation    | decimal(10,0) | YES  |     | NULL              |                             |
| RelativeHumidity | decimal(10,0) | YES  |     | NULL              |                             |
| WindGust         | decimal(10,0) | YES  |     | NULL              |                             |
| WindSpeed        | decimal(10,0) | YES  |     | NULL              |                             |
| AirTemperature   | decimal(10,0) | YES  |     | NULL              |                             |
| DewPoint         | decimal(10,0) | YES  |     | NULL              |                             |
| WindDirection    | decimal(10,0) | YES  |     | NULL              |                             |
+------------------+---------------+------+-----+-------------------+-----------------------------+
```

```
+----------+
| count(*) |
+----------+
|     1151 |
+----------+
```

This is my ttl mapping

```
<TriplesMapWindSpeed>
  a rr:TriplesMap;
  rr:logicalTable [ 
    rr:tableName  "_4UT01" 
  ];
  rr:subjectMap [    
    a rr:Subject;
    rr:termType rr:IRI;  
    #rr:termType rr:BlankNode;  
    rr:template "http://test/data/{time}";
    #rr:template "1";
    rr:class <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#MeasureData>;
  ];
  rr:predicateObjectMap [ 
      rr:predicateMap [ rr:constant <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#floatValue> ]; 
      rr:objectMap    [ rr:termType rr:Literal; rr:column "WindSpeed"; ]; 
    ];
 rr:predicateObjectMap [ 
      rr:predicateMap [ rr:constant <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#uom> ]; 
      rr:objectMap    [ rr:constant <http://knoesis.wright.edu/ssw/ont/weather.owl#milesPerHour> ]; 
    ];         
.

<TriplesMapWindSpeedObservation>
  a rr:TriplesMap;
  rr:logicalTable [ 
    rr:tableName  "_4UT01" 
  ];
  rr:subjectMap [    
    a rr:Subject;
    rr:termType rr:IRI;  
    rr:template "http://test/obs/{time}";
    rr:class <http://knoesis.wright.edu/ssw/ont/weather.owl#WindSpeedObservation>;
  ];
  rr:predicateObjectMap [ 
      rr:predicateMap [ rr:constant <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#result> ]; 
      rr:objectMap    [ rr:parentTriplesMap <TriplesMapWindSpeed> ]; 
    ];
 rr:predicateObjectMap [ 
      rr:predicateMap [ rr:constant <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#observedProperty> ]; 
      rr:objectMap    [ rr:constant <http://knoesis.wright.edu/ssw/ont/weather.owl#_WindSpeed> ]; 
    ];    
.
```

So my first issue is with an objectMap rr:constant. 

```
SELECT ?s ?o
WHERE { 
    ?s <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#uom> ?o.
}
```

This SPARQL query runs into a `ERROR [main] (MorphRDBRunner.scala:76) - Exception occured: null` exception. 

The second issue is a query like this:

```
SELECT ?val
WHERE { 
    ?s a <http://knoesis.wright.edu/ssw/ont/weather.owl#WindSpeedObservation>;
        <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#result> ?result.
    ?result <http://knoesis.wright.edu/ssw/ont/sensor-observation.owl#floatValue> ?val.
}
```

Should return the values of 1151 results but returns `INFO [main] (MorphXMLQueryResultWriter.scala:105) - 1324801 instance(s) retrieved`. 1151 x 1151 (a full join on itself?).

According to the R2RML spec, 
![screen shot 2016-01-21 at 15 57 17](https://cloud.githubusercontent.com/assets/9078335/12486210/843c2d98-c059-11e5-83bf-7776d5eea726.png)
![screen shot 2016-01-21 at 15 55 33](https://cloud.githubusercontent.com/assets/9078335/12486209/843bfbd4-c059-11e5-874e-483233c6bf11.png)
it seems like rr:constant for objectmap and rr:parentTriplesMap without join conditions are supported by the spec.

Thanks for your time! 

I used the version specified in the morph user guide https://github.com/oeg-upm/morph-rdb/wiki (3.5.15) and I am using the https://github.com/AlexDisler/mysql-vagrant box.

```
+-------------------------+
| version()               |
+-------------------------+
| 5.5.46-0ubuntu0.12.04.2 |
+-------------------------+
```

**Comment by fpriyatna** [comment]
hi Eugene, thanks for reporting this, we'll take a look of this issue.

And you are right, since the two Triples Map have the same logical table, no cross joins should be generated.

**Comment by fpriyatna** [comment]
hi Eugene, these issues should have been solved in version 3.5.16.
Please let me know if you still have any problem.

**Comment by dachafra** [comment]
issue should be solved. close issue
